src/native/thrift/protocols.ts

```diff
@@ -0,0 +1,178 @@
+    }
+    writeSetBegin(elementType: TType, size: number): void {
+        return this.concreteProtocol.writeSetBegin(elementType, size);
+    }
+    writeSetEnd(): void {
+        return this.concreteProtocol.writeSetEnd();
+    }
+    writeBool(bool: boolean): void {
+        return this.concreteProtocol.writeBool(bool);
+    }
+    writeByte(b: number): void {
+        return this.concreteProtocol.writeByte(b);
+    }
+    writeI16(i16: number): void {
+        return this.concreteProtocol.writeI16(i16);
+    }
+    writeI32(i32: number): void {
+        return this.concreteProtocol.writeI32(i32);
+    }
+    writeI64(i64: number | Int64): void {
+        return this.concreteProtocol.writeI64(i64);
+    }
+    writeDouble(dbl: number): void {
+        return this.concreteProtocol.writeDouble(dbl);
+    }
+    writeString(arg: string): void {
+        return this.concreteProtocol.writeString(arg);
+    }
+    writeBinary(arg: string | Buffer): void {
+        return this.concreteProtocol.writeBinary(arg);
+    }
+    readMessageBegin(): IThriftMessage {
+        return this.concreteProtocol.readMessageBegin();
+    }
+    readMessageEnd(): void {
+        return this.concreteProtocol.readMessageEnd();
+    }
+    readStructBegin(): IThriftStruct {
+        return this.concreteProtocol.readStructBegin();
+    }
+    readStructEnd(): void {
+        return this.concreteProtocol.readStructEnd();
+    }
+    readFieldBegin(): IThriftField {
+        return this.concreteProtocol.readFieldBegin();
+    }
+    readFieldEnd(): void {
+        return this.concreteProtocol.readFieldEnd();
+    }
+    readMapBegin(): IThriftMap {
+        return this.concreteProtocol.readMapBegin();
+    }
+    readMapEnd(): void {
+        return this.concreteProtocol.readMapEnd();
+    }
+    readListBegin(): IThriftList {
+        return this.concreteProtocol.readListBegin();
+    }
+    readListEnd(): void {
+        return this.concreteProtocol.readListEnd();
+    }
+    readSetBegin(): IThriftSet {
+        return this.concreteProtocol.readSetBegin();
+    }
+    readSetEnd(): void {
+        return this.concreteProtocol.readSetEnd();
+    }
+    readBool(): boolean {
+        return this.concreteProtocol.readBool();
+    }
+    readByte(): number {
+        return this.concreteProtocol.readByte();
+    }
+    readI16(): number {
+        return this.concreteProtocol.readI16();
+    }
+    readI32(): number {
+        return this.concreteProtocol.readI32();
+    }
+    readI64(): Int64 {
+        return this.concreteProtocol.readI64();
+    }
+    readDouble(): number {
+        return this.concreteProtocol.readDouble();
+    }
+    readBinary(): Buffer {
+        return this.concreteProtocol.readBinary();
+    }
+    readString(): string {
+        return this.concreteProtocol.readString();
+    }
+    skip(type: TType): void {
+        return this.concreteProtocol.skip(type);
+    }
+}
+
+export class TMultiplexedProtocol extends TProtocolDecorator {
```

Comment:
Based on https://github.com/apache/thrift/blob/master/lib/java/src/org/apache/thrift/protocol/TMultiplexedProtocol.java

Reply:
What's the thinking behind the intermediate `TProtocolDecorator` vs extending `TProtocol` directly?

Reply:
Discussed in person, but in summary: the ProtocolDecorator is necessary so that TMultiplexedProtocol can wrap any non-abstract TProtocol.